File: app/src/main/java/com/nodag/app/ui/WorkflowAdapter.java
package com.nodag.app.ui;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.nodag.app.R;
import com.nodag.app.workflow.WorkflowManager;
import java.util.List;

/**
 * Адаптер для списка воркфлоу
 */
public class WorkflowAdapter extends RecyclerView.Adapter<WorkflowAdapter.ViewHolder> {

    private final List<WorkflowManager> workflows;
    private final OnWorkflowClickListener listener;

    public interface OnWorkflowClickListener {
        void onWorkflowClick(WorkflowManager workflow);
    }

    public WorkflowAdapter(List<WorkflowManager> workflows, OnWorkflowClickListener listener) {
        this.workflows = workflows;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
            .inflate(R.layout.item_workflow, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(workflows.get(position));
    }

    @Override
    public int getItemCount() {
        return workflows.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        private TextView workflowName;
        private TextView workflowDescription;
        private TextView workflowNodesCount;

        ViewHolder(View itemView) {
            super(itemView);
            workflowName = itemView.findViewById(R.id.workflowName);
            workflowDescription = itemView.findViewById(R.id.workflowDescription);
            workflowNodesCount = itemView.findViewById(R.id.workflowNodesCount);
        }

        void bind(WorkflowManager workflow) {
            workflowName.setText(workflow.getName());
            workflowDescription.setText(workflow.getDescription().isEmpty() 
                ? "Нет описания" 
                : workflow.getDescription());
            workflowNodesCount.setText("Нод: " + workflow.getNodes().size());

            itemView.setOnClickListener(v -> listener.onWorkflowClick(workflow));
        }
    }
}

File: app/src/main/java/com/nodag/app/ui/WorkflowAdapter.kt
package com.nodag.app.ui

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.nodag.app.R
import com.nodag.app.workflow.WorkflowManager

/**
 * Адаптер для списка воркфлоу
 */
class WorkflowAdapter(
    private val workflows: List<WorkflowManager>,
    private val onItemClick: (WorkflowManager) -> Unit
) : RecyclerView.Adapter<WorkflowAdapter.ViewHolder>() {

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val workflowName: TextView = itemView.findViewById(R.id.workflowName)
        private val workflowDescription: TextView = itemView.findViewById(R.id.workflowDescription)
        private val workflowNodesCount: TextView = itemView.findViewById(R.id.workflowNodesCount)

        fun bind(workflow: WorkflowManager) {
            workflowName.text = workflow.name
            workflowDescription.text = workflow.description.ifEmpty { "Нет описания" }
            workflowNodesCount.text = "Нод: ${workflow.nodes.size}"

            itemView.setOnClickListener {
                onItemClick(workflow)
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_workflow, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(workflows[position])
    }

    override fun getItemCount(): Int = workflows.size
}
